### PR TITLE
Simple reddit-like voting system

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
Please expose FASTAPI endpoint.

### CODE REVIEW

File: defrag/modules/suggestions.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    @staticmethod
+    async def view(key: Optional[str] = None) -> EitherErrorOrOk:
+        if not key:
+            return await as_async(lambda: sorted(Suggestions.container.values(), key=lambda i: i["score"], reverse=True))()
+        if key and not key in Suggestions.container:
+            return EitherErrorOrOk(error=f"You were looking for this key {key} but it could not be found!")
+        return EitherErrorOrOk(ok=await as_async(lambda: Suggestions.container[key])())
+
+
+@app.get(f"/{__MODULE_NAME__}/")
```
Comment (KaratekHD) [inline defrag/modules/suggestions.py:105]:
```
[2021-09-11 11:24:19,211 | ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/fastapi/routing.py", line 216, in app
    raw_response = await run_endpoint_function(
  File "/data/jens/Development/defrag-api/venv/lib64/python3.8/site-packages/fastapi/routing.py", line 149, in run_endpoint_function
    return await dependant.call(**values)
  File "/data/jens/Development/defrag-api/defrag/modules/suggestions.py", line 109, in get_suggestions
    if ok := results.is_ok:
AttributeError: 'list' object has no attribute 'is_ok'
```

File: defrag/modules/suggestions.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        return QueryResponse(query=query, results=ok, results_count=len(ok))
+    return QueryResponse(query=query, error=results.error)
+
+
+@app.post(f"/{__MODULE_NAME__}/create/")
+async def create_suggestion(sugg: Suggestions.New) -> QueryResponse:
+    query = Query(service=__MODULE_NAME__)
+    result = await Suggestions.add(sugg)
+    if result.is_ok:
+        return QueryResponse(query=query, message="Thanks!")
```
Comment (KaratekHD) [inline defrag/modules/suggestions.py:119]: Maybe return the suggestion id here?

File: defrag/modules/suggestions.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    return QueryResponse(query=query, error="Didn't turn out the way we anticipated!")
+
+
+@app.post(f"/{__MODULE_NAME__}/vote_for_suggestion/")
+async def vote_for_suggestion(voter_id: str, sugg_id: str, vote: int) -> QueryResponse:
```
Comment (KaratekHD) [inline defrag/modules/suggestions.py:124]: How do we find out `sugg_id`?

**Review by KaratekHD — CHANGES_REQUESTED** [review]

**Comment by why-not-try-calmer** [comment]
Thank you for testing more throughly than I did, should be fixed now

**Comment by why-not-try-calmer** [comment]
Thank you!

**Pull request merged**